devito/mpi/halo_scheme.py

```diff
@@ -105,10 +106,52 @@ def fmapper(self):
         return OrderedDict([(i, self._mapper[i]) for i in
                             sorted(self._mapper, key=attrgetter('name'))])
 
+    @cached_property
+    def omapper(self):
+        """
+        Mapper describing the OWNED ('o'-mapper) region offset from the DOMAIN
+        extremes, along each Dimension and DataSide.
+
+        Examples
+        --------
+        Consider a HaloScheme comprising two one-dimensional Functions, ``u``
+        and ``v``.  ``u``'s halo, on the LEFT and RIGHT DataSides respectively,
+        is (2, 2), while ``v``'s is (4, 4). The situation is depicted below.
+
+        .. code-block:: python
+
+              xx**----------------**xx     u
+            xxxx****------------****xxxx   v
+
+        Where 'x' represents a HALO point, '*' a OWNED point, and '-' a CORE point.
+        Together, '*' and '-' constitute the DOMAIN, which is guaranteed to have
+        identical size across all Functions.
```

Comment:
Is that true? I would think this is only true if all functions are not subsampled with same Dimensions.

Reply:
in fact, it's even more subtle than that, as even w/o subsampling you might have one MPI rank in each Dimension having a slightly smaller CORE region due to the number of processes not dividing evenly the grid size

But what happened here is that the docstring is misleading. What's important is that, for each FUnction, the number of 'x' and '*' is the **same on all MPI ranks**, the '-' don't matter. And this is really true. IOW, the halo/owned regions have same size across different ranks.

I'll rephrase this docstring. Thanks for pointing it out

Reply:
I fixed the docstring